backend/src/services/auth/auth.service.ts

```diff
@@ -21,6 +23,17 @@ class Auth {
       password: await encrypt(password),
     });
   }
+
+  public async signIn(payload: SignInPayload): Promise<User | never> {
+    const { password, email } = payload;
+    const user = await this.#userRepository.getByEmail(email);
+
+    if (!user || !(await checkIsCryptsEqual (password, user.password))){
+      throw new HttpError({ status: HttpCode.NOT_FOUND, message: ErrorMessage.NOT_FOUND });
+    }
```

Comment:
```suggestion
    const user = await this.#userRepository.getByEmail(email);
    const hasUser = Boolean(user);
    
    if (!hasUser) {
       throw new HttpError({ status: HttpCode.NOT_FOUND, message: ErrorMessage.USER_NOT_FOUND });
    }
    
    const isCryptsEqual = await checkIsCryptsEqual (password, user.password);

    if (!isCryptsEqual) {
       throw new HttpError({ status: HttpCode.BAD_REQUEST, message: ErrorMessage.WRONG_PASSWORD });
    }
```

why not found is returned for password verification?